docs/api/index.rst

```diff
@@ -15,6 +15,7 @@ The following pages provide a full API reference for the ``datacube`` python lib
     core-classes/geometry
     core-classes/measurement
     core-classes/metadataType
+    core-classes/lineage-tree
```

Comment:
Minor, but would be nice for this to use camel case to be consistent with `metadataType` (or the other way around)